test/report_context_test.rb

```diff
@@ -0,0 +1,1264 @@
+  
+    # build subsequent request; should be found but miss due to freshness
+    post '/',
+      'HTTP_X_HTTP_METHOD_OVERRIDE' => 'report'
+    app.should.be.called
+    response.should.be.ok
+    response.headers.should.include 'Last-Modified'
+    response.headers.should.include 'X-Content-Digest'
+    response['Age'].to_i.should.equal 0
+    response.body.should.equal 'Hello World'
+    cache.trace.should.include :stale
+    cache.trace.should.include :valid
+    cache.trace.should.include :store
+    cache.trace.should.not.include :miss
+  end
+  
+  it 'validates cached responses with ETag and no freshness information' do
+    timestamp = Time.now.httpdate
+    respond_with do |req,res|
+      res['ETAG'] = '"12345"'
+      if req.env['HTTP_IF_NONE_MATCH'] == res['Etag']
+        res.status = 304
+        res.body = []
+      end
+    end
+  
+    # build initial request
+    post '/',
+      'HTTP_X_HTTP_METHOD_OVERRIDE' => 'report'
+    app.should.be.called
+    response.should.be.ok
+    response.headers.should.include 'ETag'
+    response.headers.should.include 'X-Content-Digest'
+    response.body.should.equal 'Hello World'
+    cache.trace.should.include :miss
+    cache.trace.should.include :store
+  
+    # build subsequent request; should be found but miss due to freshness
+    post '/',
+      'HTTP_X_HTTP_METHOD_OVERRIDE' => 'report'
+    app.should.be.called
+    response.should.be.ok
+    response.headers.should.include 'ETag'
+    response.headers.should.include 'X-Content-Digest'
+    response['Age'].to_i.should.equal 0
+    response.body.should.equal 'Hello World'
+    cache.trace.should.include :stale
+    cache.trace.should.include :valid
+    cache.trace.should.include :store
+    cache.trace.should.not.include :miss
+  end
+  
+  it 'replaces cached responses when validation results in non-304 response' do
+    timestamp = Time.now.httpdate
+    count = 0
+    respond_with do |req,res|
+      res['Last-Modified'] = timestamp
+      case (count +=1)
+      when 1 ; res.body = ['first response']
+      when 2 ; res.body = ['second response']
+      when 3
+        res.body = []
+        res.status = 304
+      end
+    end
+  
+    # first request should fetch from backend and store in cache
+    post '/',
+      'HTTP_X_HTTP_METHOD_OVERRIDE' => 'report'
+    response.status.should.equal 200
+    response.body.should.equal 'first response'
+  
+    # second request is validated, is invalid, and replaces cached entry
+    post '/',
+      'HTTP_X_HTTP_METHOD_OVERRIDE' => 'report'
+    response.status.should.equal 200
+    response.body.should.equal 'second response'
+  
+    # third respone is validated, valid, and returns cached entry
+    post '/',
+      'HTTP_X_HTTP_METHOD_OVERRIDE' => 'report'
+    response.status.should.equal 200
+    response.body.should.equal 'second response'
+  
+    count.should.equal 3
+  end
+  
+  it 'passes HEAD requests through directly on pass' do
```

Comment:
It seems like this spec should not be included

Reply:
typo in the description, but still a valid case.